app/controllers/cinema_halls_controller.rb

```diff
@@ -4,40 +4,38 @@ class CinemaHallsController < ApplicationController
   include JSONAPI::Fetching
 
   def index
-    @cinema_halls = CinemaHall.all
-    render jsonapi: @cinema_halls
+    cinema_halls = CinemaHall.all
+    render json: CinemaHalls::Representers::Multiple.new(cinema_halls).call
   end
 
   def create
-    @cinema_hall = CinemaHall.new(cinema_hall_params)
+    cinema_hall = CinemaHalls::UseCases::Create.new(params: cinema_hall_params).call
 
     if @cinema_hall.save
```

Comment:
Here is the same situation. You trying to save object itself twice. You can try to rewrite it to:
```ruby
    cinema_hall = CinemaHalls::UseCases::Create.new(params: cinema_hall_params).call
    
    if cinema_hall.valid?
      render...
    else
      render ...
    end
```

Or even better:

```ruby
    cinema_hall = CinemaHalls::UseCases::Create.new(params: cinema_hall_params).call
    
    render success
  rescue
    render failure
  end
```
But in second approach you will have to properly work with errors